algorithms/kernel/dbscan/oneapi/dbscan_dense_default_batch_ucapi_impl.i

```diff
@@ -0,0 +1,511 @@
+        if (count > maxQueueBlockSize)
+        {
+            count = maxQueueBlockSize;
+        }
+    }
+    uint32_t count;
+};
+
+template <typename algorithmFPType>
+Status DBSCANBatchKernelUCAPI<algorithmFPType>::processResultsToCompute(DAAL_UINT64 resultsToCompute, int * const isCore, NumericTable * ntData,
+                                                                        NumericTable * ntCoreIndices, NumericTable * ntCoreObservations)
+{
+    DAAL_ITTNOTIFY_SCOPED_TASK(compute.processResultsToCompute);
+    if (!isCore)
+    {
+        return Status(ErrorNullPtr);
+    }
+
+    auto & context           = Environment::getInstance()->getDefaultExecutionContext();
+    const uint32_t nRows     = ntData->getNumberOfRows();
+    const uint32_t nFeatures = ntData->getNumberOfColumns();
+
+    uint32_t nCoreObservations = 0;
+
+    for (uint32_t i = 0; i < nRows; i++)
+    {
+        if (!isCore[i])
+        {
+            continue;
+        }
+        nCoreObservations++;
+    }
+
+    if (nCoreObservations == 0)
+    {
+        return Status();
+    }
+
+    if (resultsToCompute & computeCoreIndices)
+    {
+        DAAL_CHECK_STATUS_VAR(ntCoreIndices->resize(nCoreObservations));
+        BlockDescriptor<int> indexRows;
+        DAAL_CHECK_STATUS_VAR(ntCoreIndices->getBlockOfRows(0, nCoreObservations, writeOnly, indexRows));
+        auto coreIndices = indexRows.getBuffer().toHost(ReadWriteMode::writeOnly);
+        if (!coreIndices.get())
+        {
+            return Status(ErrorNullPtr);
+        }
+
+        uint32_t pos = 0;
+        for (uint32_t i = 0; i < nRows; i++)
+        {
+            if (!isCore[i])
+            {
+                continue;
+            }
+            coreIndices.get()[pos] = i;
+            pos++;
+        }
+    }
+
+    if (resultsToCompute & computeCoreObservations)
+    {
+        DAAL_CHECK_STATUS_VAR(ntCoreObservations->resize(nCoreObservations));
+        BlockDescriptor<algorithmFPType> coreObservationsRows;
+        DAAL_CHECK_STATUS_VAR(ntCoreObservations->getBlockOfRows(0, nCoreObservations, writeOnly, coreObservationsRows));
+        auto coreObservations = coreObservationsRows.getBuffer();
+
+        uint32_t pos = 0;
+        int result   = 0;
+        for (uint32_t i = 0; i < nRows; i++)
+        {
+            if (!isCore[i])
+            {
+                continue;
+            }
+            BlockDescriptor<algorithmFPType> dataRows;
+            DAAL_CHECK_STATUS_VAR(ntData->getBlockOfRows(i, 1, readOnly, dataRows));
+            auto data = dataRows.getBuffer();
+
+            Status st;
+            context.copy(UniversalBuffer(coreObservations), pos * nFeatures, UniversalBuffer(data), 0, nFeatures, &st);
+            DAAL_CHECK_STATUS_VAR(st);
+            pos++;
+            DAAL_CHECK_STATUS_VAR(ntData->releaseBlockOfRows(dataRows));
+        }
+        if (result)
+        {
+            return Status(services::ErrorMemoryCopyFailedInternal);
+        }
+    }
```

Comment:
It would be better to divide the logic inside conditions on lines of 83 and 106 into different functions.